roofit/multiprocess/CMakeLists.txt

```diff
@@ -16,7 +16,7 @@ ROOT_LINKER_LIBRARY(RooFitMultiProcess
 )
 
 target_link_libraries(RooFitMultiProcess PUBLIC RooFitZMQ)
-set(RooFitMultiProcess_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/res ${RooFitZMQ_INCLUDE_DIRS})
+set(RooFitMultiProcess_INCLUDE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/res" ${RooFitZMQ_INCLUDE_DIRS} CACHE INTERNAL "")
```

Comment:
Why add this to the CMake cache? Usually only variables for external things which are meant to persist after configuration and be set by the user should be added to the cache.

Reply:
The problem I need to solve here is that I need to propagate the `res` include directories of MultiProcess and RooFitZMQ to `roofitcore`. I tried to do this with CMake targets, but had issues with ROOT's dictionary generation magic. Using variables was my attempt to get around that. They need to be in cache, because otherwise they cannot be found outside of their scopes (i.e. in parent/sibling directories).

If you know of a better way to solve this problem, please let me know :)

Reply:
Ok, I'm trying this out now (didn't compile yet, but config at least works):

```
target_include_directories(RooFitZMQ
        PRIVATE ${RooFitZMQ_INCLUDE_DIR}
        INTERFACE $<BUILD_INTERFACE:${RooFitZMQ_INCLUDE_DIR}>)
```

I then don't put the variable in CACHE anymore and the paths are just propagated via regular target mechanisms. Do you think this should work or do you already foresee issues?

Reply:
To elaborate: the issue that this avoids is that when I set the ${RooFitZMQ_INCLUDE_DIR} as PUBLIC in `target_include_directories`, CMake errors out saying:
```
  Target "RooFitZMQ" INTERFACE_INCLUDE_DIRECTORIES property contains path:

    "[...]/roofit-dev/root/roofit/roofitZMQ/res"

  which is prefixed in the source directory.
```
With the above solution (specifying PRIVATE and INTERFACE separately) this error disappears.

Reply:
Fixed in e90cea5 and ce08c20 (will fixup these into the other commits).